pkg/apis/monitoring/v1/prometheus_types.go

```diff
@@ -526,6 +526,48 @@ type PrometheusSpec struct {
 	// Defines the runtime reloadable configuration of the timeseries database
 	// (TSDB).
 	TSDB TSDBSpec `json:"tsdb,omitempty"`
+	// TracingConfig configures tracing in Prometheus. This is an experimental feature, it may change in any upcoming release in a breaking way.
+	// +optional
+	TracingConfig *PrometheusTracingConfig `json:"tracingConfig,omitempty"`
+}
+
+type PrometheusTracingConfig struct {
+	// Client used to export the traces. Options are "http" or "grpc".
+	//+kubebuilder:validation:Enum=http;grpc
+	//+kubebuilder:default:="grpc"
+	// +optional
+	ClientType *string `json:"clientType"`
+
+	// Endpoint to send the traces to. Should be provided in format <host>:<port>.
+	Endpoint string `json:"endpoint"`
```

Comment:
(nit) I see that it was already discussed but in order to be explicit, we could add the `//+required` annotation